localstack/services/opensearch/provider.py

```diff
@@ -368,6 +375,12 @@ def _ensure_domain_exists(arn: ARN) -> None:
         raise ValidationException("Invalid ARN. Domain not found.")
 
 
+def _transform_domain_config_request_to_status(request: Dict) -> Dict:
```

Comment:
is this the inverse operation of `_status_to_config`, if so: can we name it that way and also add the relevant typehints?

Reply:
it's not exactly an inverse. it doesn't take a DomainConfig and transforms it to a DomainStatus but takes a UpdateDomainConfigRequest and transforms it to a status. In order to able to pop the unwanted keys I need to use Dict as the typehint for the parameter here since the request t.ype wouldn't allow it. In the same vein I cannot use the DomainStatus type to do the update in L495.

I changed it now so that the function has the expected types but I transform them as soon as I need to change them.